sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/jackson/ObjectMapperTest.java

```diff
@@ -154,4 +161,47 @@ void shouldNotFailForUnknownSubtype() throws Exception {
 
     assertThat(actual).isNull();
   }
+
+  @Test
+  void shouldNotWriteValidJsonOnSerializationError() {
+    var iterator = new CloseableIterator();
+
+    var output = new ByteArrayOutputStream();
+
+    try (JsonGenerator generator = om.createGenerator(output)) {
+      generator.writeObject(Map.of("items", iterator));
+    } catch (Exception ignored) {
+      // nothing to do
+    }
+    var expected = new StringBuilder();
+    for (int i = 1; i < 1_001; i++) {
+      expected.append("\"").append(i).append("\",");
+    }
+    expected.deleteCharAt(expected.length() - 1);
+    assertThat(output.toString(UTF_8)).isEqualToIgnoringWhitespace("{\"items\":[" + expected);
+  }
+
+  static class CloseableIterator implements Iterator<String>, Closeable {
```

Comment:
Better name `FailingIterator` to make sure what the class should do?

Reply:
changed to `FailingClosableIterator`